include/platform/messagepool.h

```diff
@@ -63,14 +63,13 @@ extern "C" {
 #endif
 
 /**
- * Initialize the platform implemented message pool with the provided buffer list.
+ * Initialize the platform implemented message pool.
  *
- * @param[in] aFreeBufferList   A linked list of free buffers.
- * @param[in] aNumFreeBuffers   A pointer to an int containing the number of free buffers in aFreeBufferList.
- * @param[in] aBufferSize       The size in bytes of a Buffer object.
+ * @param[in] aMinNumFreeBuffers   An int containing the minimum number of free buffers desired by OpenThread.
+ * @param[in] aBufferSize          The size in bytes of a Buffer object.
  *
  */
-void otPlatMessagePoolInit(struct BufferHeader *aFreeBufferList, int *aNumFreeBuffers, size_t aBufferSize);
+void otPlatMessagePoolInit(int aMinNumFreeBuffers, size_t aBufferSize);
```

Comment:
All this looks great, but why even have a notion of 'number of buffers', either for initial count or for a current count? Why not just support dynamically allocating/freeing and if an allocation fails treat that as out of buffers?